src/mon/Monitor.cc

```diff
@@ -418,7 +418,17 @@ int Monitor::do_admin_command(
       cmd_vec.push_back(val);
     }
     ceph_heap_profiler_handle_command(cmd_vec, out);
-  } else {
+  } else if (command == "compact" || command == "mon compact") {
```

Comment:
let's drop 'mon compact' and keep just 'compact'.  for these commands it's not so important to maintain backward compatibility

Reply:
@liewegas i am reading #4595, if we instead drop "compact". that would defeat the purpose of @jecluis 's original intention. shall we just drop "compact" and "scrub", and keep "mon compact" and "mon scrub"?

Reply:
This is a mon-specific command--it doesn't make sense as a CLI command for the entire cluster--it only makes sense as a command to tell a specific monitor.  Like ``ceph tell mon.a compact``.  Back when @jecluis did #4595 these commands were all mixed together and putting it under 'ceph mon ...' made sense, but now you're specifially sending it to a mon, so the 'mon' part of the command is redundant.